EIPS/eip-3675.md

```diff
@@ -0,0 +1,246 @@
+
+To understand the motivation of introducing the Proof-of-Stake consensus see the Motivation section of [EIP-2982](/EIPS/eip-2982#motivation).
+
+
+## Specification
+
+### Definitions
+
+* **PoW block**: Block that is built and verified by the existing proof-of-work mechanism. In other words, a block of the Ethereum network before the consensus upgrade.
+* **PoS block**: Block that is built and verified by the new proof-of-stake mechanism.
+* **Terminal PoW block**: A PoW block that satisfies the following conditions -- 
+`pow_block.total_difficulty >= TERMINAL_TOTAL_DIFFICULTY` *and* `pow_block.parent_block.total_difficulty < TERMINAL_TOTAL_DIFFICULTY`.  
+There can be more than one terminal PoW block in the network, e.g. multiple children of the same pre-terminal block.
+* **`TERMINAL_TOTAL_DIFFICULTY`** The amount of total difficulty reached by the network that triggers the consensus upgrade.
+* **`TRANSITION_BLOCK`** The earliest PoS block of the canonical chain, i.e. the PoS block with the lowest block height.  
+`TRANSITION_BLOCK` **MUST** be a child of a terminal PoW block.
+* **`POS_CONSENSUS_VALIDATED`** An event occurring when the PoS block is validated with respect to the proof-of-stake consensus rules.
+* **`POS_CHAINHEAD_SET`** An event occurring when the PoS block is designated as the head of the canonical chain.
+* **`POS_BLOCK_FINALIZED`** An event occurring when the PoS block is considered as finalized.
+
+#### PoS events
+
+Events having the `POS_` prefix in the name (PoS events) are emitted by the new proof-of-stake consensus mechanism. They signify the corresponding assertion that has been made regarding a block specified by the event. The underlying logic of PoS events can be found in the beacon chain specification. On the occurrence of each PoS event the corresponding action that is specified by this EIP **MUST** be taken.
+
+Each `POS_BLOCK_FINALIZED` event has a reference to the head of the canonical chain in addition to the finalized block and incorporates the semantics of the `POS_CHAINHEAD_SET` event. Thus, every action and effect that is specified on the occurrence of `POS_CHAINHEAD_SET` event according to this EIP **MUST** also be taken on every occurrence of `POS_BLOCK_FINALIZED` event.
+
+
+### PoW block processing
+
+PoW blocks that are descendants of any terminal PoW block **MUST NOT** be imported. This implies that a terminal PoW block will be the last PoW block in the canonical chain.
+
+
+### Block structure
+
+Beginning with `TRANSITION_BLOCK`, a number of previously dynamic block fields are deprecated by enforcing these values to instead be constants. Each block field listed in the table below **MUST** be replaced with the corresponding constant value.
+
+| Field | Constant value |
+|-|-|
+| **`ommersHash`** | `Keccak256(RLP(EMPTY_LIST))`  |
+| **`difficulty`** | `0` |
+| **`extraData`** | `EMPTY_LIST` |
```

Comment:
`extraData` is a `bytes`, no? So if  you want it empty, it should not be an rlp-list, IIRC, but rather a `0`-length byte array

Reply:
According to https://eth.wiki/fundamentals/specs/consensus-datastructures, which may be wrong, it is a `UInt256`, which is serialized as a leading-zero-truncated `byte[]`.  If that linked page is wrong we should update it as well as this EIP.

Reply:
Good catch! From YP: 
> **extraData**: An arbitrary byte array containing data relevant to this block.

Reply:
Hmm, would it be better to either replace `EMPTY_LIST` with `0xc0` or replace `0x80` with `EMPTY_BYTE_ARRAY`? Having `EMPTY_LIST` aside with `0x80` is a bit messy.

Reply:
The correct empty value `0x80`, so that should be `EMPTY_BYTE_ARRAY`.